--- php-5.4.11.orig/sapi/cli/php_cli.c	2013-01-16 07:10:30.000000000 +0000
+++ php-5.4.11/sapi/cli/php_cli.c	2013-04-25 14:54:11.000000000 +0000
@@ -454,7 +454,7 @@
 
 	php_error,						/* error handler */
 
-	sapi_cli_header_handler,		/* header handler */
+	NULL,		/* header handler */
 	sapi_cli_send_headers,			/* send headers handler */
 	sapi_cli_send_header,			/* send header handler */
